kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/AdminResourceSuite.scala

```diff
@@ -470,6 +520,57 @@ class AdminResourceSuite extends KyuubiFunSuite with RestFrontendTestHelper {
     }
   }
 
+  test("list engine - group share level with SimpleGroupProvider") {
+    val id = UUID.randomUUID().toString
+    conf.set(KyuubiConf.ENGINE_SHARE_LEVEL, GROUP.toString)
+    conf.set(KyuubiConf.ENGINE_TYPE, SPARK_SQL.toString)
+    conf.set(KyuubiConf.FRONTEND_THRIFT_BINARY_BIND_PORT, 0)
+    conf.set(HighAvailabilityConf.HA_NAMESPACE, "kyuubi_test")
+    conf.set(KyuubiConf.ENGINE_IDLE_TIMEOUT, 180000L)
+    conf.set(KyuubiConf.GROUP_PROVIDER, "org.apache.kyuubi.server.api.v1.SimpleGroupProvider")
+    conf.set("kyuubi.engine.group.name", "second")
+
+    val groupProvider = PluginLoader.loadGroupProvider(conf)
+
+    val engine =
+      new EngineRef(conf.clone, Utils.currentUser, groupProvider, id, null)
+
+    val engineSpace = DiscoveryPaths.makePath(
+      s"kyuubi_test_${KYUUBI_VERSION}_GROUP_SPARK_SQL",
+      groupProvider.primaryGroup(
+        Utils.currentUser,
+        conf.getAll.asJava),
+      "")
+
+    withDiscoveryClient(conf) { client =>
+      engine.getOrCreate(client)
+
+      assert(client.pathExists(engineSpace))
+      assert(client.getChildren(engineSpace).size == 1)
+
+      val response = webTarget.path("api/v1/admin/engine")
+        .queryParam("type", "spark_sql")
+        .queryParam("sharelevel", "GROUP")
+        .queryParam("groupname", "second")
+        .request(MediaType.APPLICATION_JSON_TYPE)
+        .header(AUTHORIZATION_HEADER, s"BASIC $encodeAuthorization")
+        .get
```

Comment:
Hi @pan3793 could you please help me on how to change the `GroupProvider` of the server that the `get` request send to?  It currently the uses `fe.sessionManager.groupProvider` which is the hadoop group provider, right.